code/modules/robotics/bot/duckbot.dm

```diff
@@ -124,6 +124,21 @@
 		if(frustration >= 8)
 			src.KillPathAndGiveUp(1)
 
+/obj/machinery/bot/duckbot/attack_ai(var/mob/user as mob)
+	if(!ON_COOLDOWN(src,"ai_quack", 1 SECOND))
+		var/quack_now = TRUE
+		var/quack_time_remaining = GET_COOLDOWN(src, DUCKBOT_QUACK_COOLDOWN)
+		if(quack_time_remaining)
+			if(prob(66))
+				src.cooldowns[DUCKBOT_QUACK_COOLDOWN] = 0
+				quack_now = FALSE
```

Comment:
Why reset the cooldown here? Or are you using this to simulate a random delay for spam clicking maybe?

Reply:
You got it. 33% of the time it will quack immediately OR it will quack faster than it would without AI interaction.  Makes the quacking a little bit less 66% less spammy if the duck just quacked naturally.

It causes there always to be a benefit to clicking the duck (outside the 1 second click cooldown).  Just sometimes you have to wait for the duck to machine loop to trigger it.

From a maintainability standpoint I can see that it kinda sucks given that this is a lot of unusual computations and adjustments to values that are typically simply abstracted by the cooldown structure.